### ISSUE TITLE
rancher-compose support for profiles/environments

### ISSUE THREAD

**Issue by johnrengelman** [rancher/rancher#3311]
Currently it's required to have `RANCHER_URL`, `RANCHER_ACCESS_KEY` and `RANCHER_SECRET_KEY` specified when using rancher-cli. 
It would be nice to be able to specify these as part of a `.rancherrc` file or something similar. `rancher-compose` could then have a `--environment` flag that would load those values from the rc file. 
Additionally, the `environment` could be specified as part of the `rancher-compose.yml` file and used as a default.

This would facilitate using version control for stack configuration.

**Comment by RVN-BR** [comment]
Would this be in addition the env variables already supported by rancher-compose?

```
# Set the url that Rancher is on
$ export RANCHER_URL=http://server_ip:8080/
# Set the access key, i.e. username
$ export RANCHER_ACCESS_KEY=<username_of_key>
# Set the secret key, i.e. password
$ export RANCHER_SECRET_KEY=<password_of_key>
```
you can have simple init.sh file you run before your rancher-compose commands if thats for convenience?

Having the data in the rancher-compose file could be nice tho...makes sense imo too, as that would define the target of the stack in the rancher-compose.yml file...

**Comment by johnrengelman** [comment]
Yes, this would be in addition to the envvars.
A wrapper script is a bit how i've been running, but it would be great for first class support for this.

**Comment by cusspvz** [comment]
+1 for `.rancherrc`

**Comment by fernandoneto** [comment]
+1 in `.rancherrc`

**Comment by RVN-BR** [comment]
Sorry about that, my bad, I missed looking at the documentation before speaking...

http://docs.rancher.com/rancher/rancher-compose/commands/

under options:
```
--env-file, -e	Specify a file from which to read environment variables
```

I'm not sure the formatI guess it should be fairly straight fwd... I tried a couple of formats, however, and didnt get it to work... so I think clarification is needed abt the format... (tried json, yml, and using exports, or just VAR=VALUE one per line... all failed with "RANCHER_URL is not set")

On a separate note, but seen in that page, the -p option is a bit weirdly worded... Rancher makes references to STACKS, and not projects, maybe its some old wording?

**Comment by johnrengelman** [comment]
@RVN-BR I don't believe that's the same thing. My guess is that it is a file that will be sourced and then interpolated into the `docker-compose.yml` and `rancher-compose.yml` files.

Either way, that would still have the same problem in the the user is responsible for effectively deciding which environment to put the stack in.

What I want, is to explicitly say, "I expect this stack to be in the environment (profile). Go get my credentials for that environment (profile) from a know spot (.rancherrc) and use them when running any commands for this stack"

**Comment by RVN-BR** [comment]
Hi @johnrengelman now that you mention that, it does make sense... (and explains why its not working in my tests)...  Its grabbing an env file for docker-compose, and not for itself...  Oh well... I'm using a simple sh script for now, working for me...

in case anyone is interested...
rcp.sh (dont forget to 755)
```
#!/bin/bash
export RANCHER_URL=http://172.17.8.101:8080/
export RANCHER_ACCESS_KEY=02519EDDBDDFEAAEDB67
export RANCHER_SECRET_KEY=WxgT6iyJ5eoPwQBbHNy1Gp2AmKZD4Rua3QFgpjQR
./rancher-compose "$@"
```

**Comment by wjimenez5271** [comment]
should this be merged #7075 since rancher cli will become the standard CLI tool?

**Comment by deniseschannon** [comment]
With the release of Rancher 2.0, development on v1.6 is only limited to critical bug fixes and security patches.
